presto-main-base/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -268,11 +266,16 @@ public DeleteNode plan(Delete node)
         }
 
         // add rowId column
-        Field rowIdField = Field.newUnqualified(node.getLocation(), Optional.empty(), rowIdType);
-        VariableReferenceExpression rowIdVariable = variableAllocator.newVariable(getSourceLocation(node), "$rowId", rowIdField.getType());
-        outputVariablesBuilder.add(rowIdVariable);
-        columns.put(rowIdVariable, rowIdHandle);
-        fields.add(rowIdField);
+        Optional<ColumnHandle> rowIdHandle = metadata.getDeleteRowIdColumn(session, handle);
+        Optional<Field> rowIdField = Optional.empty();
+        if (rowIdHandle.isPresent()) {
+            Type rowIdType = metadata.getColumnMetadata(session, handle, rowIdHandle.get()).getType();
+            rowIdField = Optional.of(Field.newUnqualified(node.getLocation(), Optional.empty(), rowIdType));
+            VariableReferenceExpression rowIdVariable = variableAllocator.newVariable(getSourceLocation(node), "$rowId", rowIdField.get().getType());
```

Comment:
```suggestion
            VariableReferenceExpression rowIdVariable = variableAllocator.newVariable(getSourceLocation(node), "$rowId", rowIdType);
```

nit: we can directly use `rowIdType`